### PR TITLE
Sorting room list alphabetically case-insensitive.

### PR DESCRIPTION
**Pull request by NicolasBuquet**
When sorting room list alphabetically, sort it case-insensitive.

Simulator Screenshot - iPhone 15 - 2024-05-07 at 11 58 20

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request contains a changelog file in ./changelog.d. See https://github.com/matrix-org/matrix-ios-sdk/blob/develop/CONTRIBUTING.md#changelog
* [x] Pull request includes a [sign off](https://github.com/matrix-org/matrix-ios-sdk/blob/develop/CONTRIBUTING.md#sign-off)

### CODE REVIEW

**Comment by NicolasBuquet** [comment]
@pixlwave Can you have a look please. 🙏

**Review by pixlwave — COMMENTED** [review]
Oh, thanks thats a nice little improvement. It might be better to use [`localizedStandardCompare`](https://developer.apple.com/documentation/foundation/nsstring/1409742-localizedstandardcompare) here instead though, so the sorting is aligned with other places on the system?

**Comment by NicolasBuquet** [comment]
@pixlwave 
You're right. I tested and [localizedStandardCompare](https://developer.apple.com/documentation/foundation/nsstring/1409742-localizedstandardcompare) is case-insensitive, so I did the asked modification.

Thanks 👍

File: changelog.d/sort-room-list-alphabetically-case-insensitive.change
Comment (pixlwave) [inline changelog.d/sort-room-list-alphabetically-case-insensitive.change:1]: As mentioned in the [guidelines](https://github.com/matrix-org/matrix-ios-sdk/blob/develop/CONTRIBUTING.md#changelog), this file name should either be the issue number, or in this instance as there isn't one, `pr-1851.change`.

**Review by pixlwave — APPROVED** [review]
Looks good to me, just one tweak on the changelog and we can merge this.

File: changelog.d/sort-room-list-alphabetically-case-insensitive.change
Comment (pixlwave) [reply changelog.d/sort-room-list-alphabetically-case-insensitive.change:1]: Nice, thanks very much 🙏

**Pull request merged**